internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go

```diff
@@ -120,73 +119,68 @@ func resourceCassandraDatacenter() *pluginsdk.Resource {
 }
 
 func resourceCassandraDatacenterCreate(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Cosmos.CassandraDatacentersClient
+	client := meta.(*clients.Client).Cosmos.ManagedCassandraClient
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	clusterId, _ := parse.CassandraClusterID(d.Get("cassandra_cluster_id").(string))
-	id := parse.NewCassandraDatacenterID(clusterId.SubscriptionId, clusterId.ResourceGroup, clusterId.Name, d.Get("name").(string))
+	clusterId, _ := managedcassandras.ParseCassandraClusterID(d.Get("cassandra_cluster_id").(string))
```

Comment:
line 60 (above) is validating this with a different ID parser, so it's possible there could be an error here - can we both update the schema to check the validation function and raise this error (we should _always_ be surfacing errors)

Reply:
Updated